### ISSUE TITLE
Got some problem when training?

### ISSUE THREAD

**Issue by MrSeven77** [h-shahidi/2birds-gen#2]
Thanks for sharing the code.
However, I got some problem when running the code on SQUAD split-2 dataset.
After running for 4 epochs, the dev loss and dev accu remain the unchanged for the following epochs.
![image](https://user-images.githubusercontent.com/42609921/90598197-2cb5ca00-e225-11ea-8554-b80f68bddd67.png)
The training screenshot is above.
I'm running on tensorflow1.12 version plaform, and I deliverately changed the code from python2 to python3.(with some minor bug fix) I wonder whether this could affect the training process. 
Little advice would be appreciated.

**Comment by h-shahidi** [comment]
I haven't tried training with python 3. Do you have the same problem with python 2?

**Comment by MrSeven77** [comment]
Hi,
I tried on python2, but get similar problems that the training loss is too large from convergence. Here's the training log:
```
<utils.config_utils.objdict instance at 0x7fd010f249b0>
Step 8663: loss = 975891.76 (2589.900 sec)
Dev loss = 101256.6008
Dev accu = 0.0515 5719/111100
Duration 93.724 sec
Step 17327: loss = 975825.27 (2675.226 sec)
Dev loss = 101366.8211
Dev accu = 0.0507 5638/111100
Duration 89.254 sec
Step 25991: loss = 975971.48 (2669.724 sec)
Dev loss = 101330.7209
Dev accu = 0.0519 5767/111100
Duration 98.967 sec
Step 34655: loss = 975707.15 (2678.237 sec)
Dev loss = 101385.7476
Dev accu = 0.0505 5614/111100
Duration 89.480 sec
Step 43319: loss = 975850.75 (2670.103 sec)
Dev loss = 101407.3972
Dev accu = 0.0444 4930/111100
Duration 89.254 sec
Step 51983: loss = 976002.57 (2671.218 sec)
Dev loss = 101422.1141
Dev accu = 0.0492 5470/111100
Duration 89.485 sec
Step 60647: loss = 976005.86 (2671.845 sec)
Dev loss = 101419.5878
Dev accu = 0.0517 5748/111100
Duration 89.650 sec
Step 69311: loss = 976005.19 (2669.480 sec)
Dev loss = 101426.1928
Dev accu = 0.0515 5725/111100
Duration 89.218 sec
Step 77975: loss = 976006.26 (2669.144 sec)
Dev loss = 101378.8662
Dev accu = 0.0518 5751/111100
Duration 89.083 sec
Step 86639: loss = 976001.47 (2670.488 sec)
Dev loss = 101363.8650
Dev accu = 0.0511 5677/111100
Duration 89.168 sec
Step 95303: loss = 976006.16 (2668.373 sec)
Dev loss = 101422.0574
Dev accu = 0.0518 5752/111100
Duration 89.847 sec
Step 103967: loss = 975944.01 (2669.511 sec)
Dev loss = 101436.7390
Dev accu = 0.0502 5576/111100
Duration 89.522 sec
Step 112631: loss = 975853.29 (2669.108 sec)
Dev loss = 102307.7765
Dev accu = 0.0413 4584/111100
Duration 89.444 sec
Step 121295: loss = 976005.34 (2667.027 sec)
Dev loss = 101361.8283
Dev accu = 0.0524 5820/111100
Duration 99.350 sec
Step 129959: loss = 976004.76 (2676.894 sec)
Dev loss = 101439.1283
Dev accu = 0.0495 5500/111100
Duration 89.379 sec
Step 138623: loss = 975851.98 (2669.460 sec)
Dev loss = 101406.3111
Dev accu = 0.0490 5444/111100
Duration 89.109 sec
Step 147287: loss = 975683.98 (2669.719 sec)
Dev loss = 101327.0549
Dev accu = 0.0531 5894/111100
Duration 100.088 sec
Step 155951: loss = 975684.53 (2680.751 sec)
Dev loss = 101405.0307
Dev accu = 0.0499 5546/111100
Duration 89.522 sec
```

**Comment by h-shahidi** [comment]
Are you using the same Tensorflow version? Have you made any change to the code?

**Comment by MrSeven77** [comment]
Thanks for your quick reply.
Yes, I follow the README and use the environment bellow:
```
Tensorflow==1.10.0
Numpy==1.14.5
NLTK==3.2(since the latest 3.5 version dropped support for python2)
```
I downloaded the split2 dataset from the url provided, and edit config.json according to the paper:
```
encoder/decoder_hidden_size=512
dropout_rate=0.3
batch_size=10 (since bigger batch size would casue OOM problem)
```
Is that because the different batch size  that cause such problem?
12GB's memory can't afford bigger batch size.

**Comment by h-shahidi** [comment]
If this is the only thing that differs, I would get suspicious about it.
